airgun/entities/discoveryrule.py

```diff
@@ -0,0 +1,157 @@
+        view.flash.assert_no_error()
+        view.flash.dismiss()
+
+    def enable(self, entity_name):
+        """Enable corresponding Discovery rule
+
+        :param str entity_name: name of the corresponding discovery rule
+        """
+        view = self.navigate_to(self, 'All')
+        view.table.row(name=entity_name)['Actions'].widget.fill('Enable')
+        self.browser.handle_alert()
+        view.flash.assert_no_error()
+        view.flash.dismiss()
+
+    def disable(self, entity_name):
+        """Disable corresponding Discovery rule
+
+        :param str entity_name: name of the corresponding discovery rule
+        """
+        view = self.navigate_to(self, 'All')
+        view.table.row(name=entity_name)['Actions'].widget.fill('Disable')
+        self.browser.handle_alert()
+        view.flash.assert_no_error()
+        view.flash.dismiss()
+
+    def associated_hosts(self, entity_name, host_name):
+        """Associate host to corresponding Discovery rule
+
+        :param str entity_name: name of the discovery rule
+        :param host_name: name of the host which is associated to disc. rule
+        :return: list of table rows of Host entity, each row is dict,
+            where attribute is key with correct value
+        """
+        view = self.navigate_to(self, 'All')
+        view.table.row(
+            name=entity_name)['Actions'].widget.fill('Associated Hosts')
+        view = self.navigate_to(self, 'Associate', host_name=host_name)
+        return view.table.read()
+
+
+@navigator.register(DiscoveryRuleEntity, 'All')
+class ShowAllDiscoveryRules(NavigateStep):
+    """Navigate to All Discovery rules screen."""
+    VIEW = DiscoveryRulesView
+
+    def step(self, *args, **kwargs):
+        self.view.menu.select('Configure', 'Discovery rules')
+
+
+@navigator.register(DiscoveryRuleEntity, 'New')
+class AddNewDiscoveryRule(NavigateStep):
+    """Navigate to Create Discovery rule page."""
+    VIEW = DiscoveryRuleCreateView
+
+    prerequisite = NavigateToSibling('All')
+
+    def step(self, *args, **kwargs):
+        self.parent.browser.click(self.parent.new)
+
+
+@navigator.register(DiscoveryRuleEntity, 'Edit')
+class EditDiscoveryRule(NavigateStep):
+    """Navigate to Edit Discovery rule page.
+
+        Args:
+            entity_name: name of the discovery rule
+    """
+    VIEW = DiscoveryRuleEditView
+
+    def prerequisite(self, *args, **kwargs):
+        return self.navigate_to(self.obj, 'All')
+
+    def step(self, *args, **kwargs):
+        entity_name = kwargs.get('entity_name')
+        self.parent.table.row(name=entity_name)['Name'].widget.click()
+
+
+@navigator.register(DiscoveryRuleEntity, 'Associate')
+class AssociateHosts(NavigateStep):
+    """Navigate to Associated hosts page.
+
+        Args:
+            host_name: name of the host, which is associated
+    """
+    VIEW = HostsView
+
+    def am_i_here(self, *args, **kwargs):
+        return self.view.is_displayed and self.view.title == 'Hosts'
```

Comment:
`self.view.is_displayed` is actually verify that `title` is present due
```
title = Text("//h1[text()='Hosts']")

@property
def is_displayed(self):
    return self.browser.wait_for_element(
        self.title, exception=False) is not None
```
so ```am_i_here``` looks a little bit redundant
especially taking into account that it will never work as `self.view.title` is a Widget, not a string

How does `Associate Hosts` functionality works?
It is actually go on a Host page with already filled search bar like `discovery_rule = "testdr"`, so `step` is also looks wrong. Probably, it should be like:
```
def am_i_here(self, *args, **kwargs):
    return self.view.is_displayed and self.view.searchbar.read() == 'discovery_rule = rule_name`

def step
    => to all hosts
    search(discovery_rule = rule_name)
```
but I don't like to have step at all as if you don't get to necessary page from first attempt, retry mechanism of Navmazing can mask a possible bug in the application